### PR TITLE
Complaint tool schema

### PR DESCRIPTION
**Pull request by cehsu**
These changes set up the complaint tool schema.

Outstanding questions have been noted as PR comments, and as inline comments, including those relating to discrepancies between the provided list, the example request, and the design (e.g. level of study).

TODO:
- [x] tests
- [x] bump version
- [x] build dist

### CODE REVIEW

File: src/schemas/complaint-tool/schema.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      activeICN: {
+        type: 'number'
+      },
+      historicalICN: { // TODO: should this be "historicalICNs" (plural)?
+        type: 'array',
+        items: {
+          type: 'object',
+          properties: {
+            ICN: {
+              type: 'number'
```
Comment (cehsu) [inline src/schemas/complaint-tool/schema.js]: Are these typically type number?

File: src/schemas/complaint-tool/schema.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      type: 'string',
+      maxLength: 1000
+    }
+  }
+};
+
+[
+  ['privacyAgreementAccepted'], // should this be removed on the BE?
+  ['usaPhone', 'phone'],
+  ['address'],
```
Comment (cehsu) [inline src/schemas/complaint-tool/schema.js]: Do we want to copy how VIC defines address here?

File: src/schemas/complaint-tool/schema.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      ]
+    },
+    fullName: {
+      '$ref': '#/definitions/fullName'
+    },
+    email: {
+      type: 'string',  // TODO: determine if there is a length limit
+      format: 'email'
+    },
+    profileData: {
```
Comment (lihanli) [inline src/schemas/complaint-tool/schema.js]: we don't need to include this in the schema, it won't be used in the frontend

File: src/schemas/complaint-tool/schema.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+            'Mrs.',
+            'Ms.',
+            'Dr.',
+            'Other'
+          ]
+        }
+      }
+    })
+  },
+  required: [
```
Comment (cehsu) [inline src/schemas/complaint-tool/schema.js]: @lihanli how do we typically handle conditional requirements? For example name is only required if "anonymous" is not selected? Is this only handled on the FE? [Schema draft 7 includes if else support](https://ajv.js.org/keywords.html#ifthenelse), should we consider using that?

Comment (lihanli) [reply src/schemas/complaint-tool/schema.js]: something like this should work
```javascript
{
  oneOf: [
    {
      properties: {
        onBehalfOf: {
          type: 'string',
          'enum': [
            'Myself',
            'Someone else'
          ]
        },
      }
      required: ['fullName']
    },
    {
      properties: {
        onBehalfOf: {
          type: 'string',
          'enum': [
            'Anonymous'
          ]
        },
      }
    }
  ]
}
```

Comment (cehsu) [reply src/schemas/complaint-tool/schema.js]: Thanks @lihanli .

File: src/schemas/complaint-tool/schema.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+      format: 'email'
+    },
+    educationDetails: {
+      required: [ // no fields are required for submission, though several are required by design on FE
+        // 'school',
+        // 'programs'
+      ],
+      school: definitions.school,
+      programs: { // TODO: Needs to be translated into a (comma) delimited string, and spelled out if within 255 limit 
+        type: 'object',
```
Comment (rianfowler) [inline src/schemas/complaint-tool/schema.js]: Maybe could use an array of enum here.

Comment (cehsu) [reply src/schemas/complaint-tool/schema.js]: Thanks @rian, these will be presented as checkboxes, which we typically create using booleans.

File: src/schemas/complaint-tool/schema.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+      type: 'string',
+      maxLength: 1000
+    }
+  }
+};
+
+[
+  ['privacyAgreementAccepted'],
+  ['usaPhone', 'phone'], // TODO: note validation requirements pending feedback
+  ['address'], // TODO: note length limits pending feedback
```
Comment (lihanli) [inline src/schemas/complaint-tool/schema.js]: did we get an answer about international addresses? seems like we should use the same address definition that's used in the vic schema

Comment (cehsu) [reply src/schemas/complaint-tool/schema.js]: Actually, only US addresses will be accepted, both the school and the applicant address have different validations as well.

**Comment by cehsu** [comment]
Outstanding questions, a list of translations has been added to the [wiki](https://github.com/department-of-veterans-affairs/vets.gov-team/wiki/Rainbows-2018-Technical-Documentation).

**Pull request merged**